master/pkg/tasks/task.go

```diff
@@ -261,7 +287,10 @@ func injectUserArchive(aug *model.AgentUserGroup, workDir string) cproto.RunArch
 	passwdBytes := []byte(
 		fmt.Sprintf("%v:x:%v:%v::%v:/bin/bash\n", aug.User, aug.UID, aug.GID, workDir),
 	)
-	shadowBytes := []byte(fmt.Sprintf("%v:!!:::::::\n", aug.User))
+	// Add a bogus password in the shadow file.  Cannot use ! as that locks the account
+	// when using SLURM/Singularity.   The password is not actually ever used so OK that
+	// it can never be specified due to our sshd_config is configured with "PasswordAuthentication no"
+	shadowBytes := []byte(fmt.Sprintf("%v:BogusPasswordHash:::::::\n", aug.User))
```

Comment:
maybe @rb-determined-ai should review these lines, since he wrote our libnss stuff.

Reply:
Cat we use `*` instead of `BogusPasswordHash`?

I see from `man 5 shadow` that `*` or `!` would both cause password logins to fail, though `!` has the special meaning that the account is locked (which I suppose something in the slurm/singularity stack is looking at).

Reply:
Right.  !  was completely blocked.   I can try * instead if you prefer, but gonna delay this til next week, since I can't test on this branch.   I can only test this support on the determined-ee dispatcher branch at the moment.

Reply:
It wasn't actually that hard to test -- so pushed up the change to use * instead.